### ISSUE TITLE
Module build failed: TypeError: Property right of AssignmentExpression expected node to be of a type ["Expression"] but instead got null at Array.map (native)

### ISSUE THREAD

**Issue by BigFaceMaster** [babel/babel#7831]
Choose one: is this a bug report or feature request?

### Input Code

```js
var your => (code) => here;
```

### Babel/Babylon Configuration (.babelrc, package.json, cli command)

```js
{
  "your": { "config": "here" }
}
```

### Your Environment

| software         | version(s)
| ---------------- | -------
| Babel            |  
| Babylon          | 
| node             | 
| npm              | 
| Operating System |

**Comment by nicolo-ribaudo** [comment]
Please fill the issue template

**Comment by cburbank** [comment]
@leejay1992 I started seeing this error after upgrading my project to babel beta 46 (was previously on beta 44), and it seems that it may indicate some conflict with babel plugins. In my case I was able to resolve it by removing `babel-plugin-transform-decorators-legacy` and adding `"decoratorsLegacy": true` to the `@babel/env` preset options.

**Comment by BingoBinBingo** [comment]
Good Nice

**Comment by somonek** [comment]
@cburbank's solution didn't work for me, but I was able to solve it by having the decorators plugin like this:
```
{
  "presets": [
    "@babel/preset-react",
    ["@babel/preset-env", {
      "targets": {
        "browsers": ["last 2 versions"]
      }
    }]
  ],
  "plugins": [
    ["@babel/plugin-proposal-decorators", {
      "legacy": true
    }],
    "react-hot-loader/babel",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-object-rest-spread",
    "@babel/plugin-syntax-export-extensions",
    "transform-function-bind"
  ]
}

```

**Comment by capaj** [comment]
@somonek thanks for your comment-I've also resolved this issue by replacing the plugin:

![image](https://user-images.githubusercontent.com/1305378/39917482-f1b469f8-550d-11e8-923b-2acf6a60e151.png)

**Comment by dooboolab** [comment]
![image](https://user-images.githubusercontent.com/27461460/42313848-a59383fa-807e-11e8-8c7d-1bca3c3805e9.png)
I am suffering from this issue in `react-native@0.56` any idea?

**Comment by TitanKing** [comment]
@dooboolab Did you manage to find a solution for this?

**Comment by mCodex** [comment]
I've also tried:

```
  "resolutions": {
    "babel-core": "^7.0.0-bridge.0"
  },
```
in my package.json and then:

```bash
rm -rf node_modules/ yarn.lock && yarn
```

but still doesn't work :/

**Comment by dimapaloskin** [comment]
Looks like this bug related with this commit: https://github.com/babel/babel/commit/b8dcd6f593955be2779b770182a4e8930288cdaa
which totally renamed `callee` to `expression`

The quick fix:
https://github.com/babel/babel/blob/master/packages/babel-plugin-proposal-decorators/src/transformer-legacy.js#L49

`const expression = decorator.expression;` -> `const expression = decorator.expression || decorator.callee`

**Comment by nicolo-ribaudo** [comment]
@dimapaloskin That is only a problem when you are using different versions of the various packages.

**Comment by xuqinggang** [comment]
Note: please ensure babel plugin package version is consistent!
![image](https://user-images.githubusercontent.com/13174560/42570853-d4ad84c4-8547-11e8-8604-5d93083672b2.png)

**Comment by nicolo-ribaudo** [comment]
You shouldn't use `^` in beta versions: they could get out of sync

**Comment by dooboolab** [comment]
@rknell Thanks. My app is running now. However, there is another problem left for me which is jest. It works fine when I downgrade to `react-native@0.55.4` but throws error message in `0.56`.

```
 Requires Babel "^7.0.0-0", but was loaded with "6.26.3". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel.
```

If you want to checkout my boilerplate I am working on,  it is in [here](https://github.com/react-native-seoul/react-native-js-boilerplate/tree/dev). Be sure that it is a `dev` branch. I'll merge to master when everything is working fine.

Thank you by the way with the `decorator` issue.

**Comment by hvaoc** [comment]
I got the same error when adding Mobx support to my react native project. 

Changes suggested by @dooboolab above fixed this error for me.

Am using "react-native": "^0.56.0".

**Comment by mehulmpt** [comment]
Yes. Downgrading to 0.47 version fixes the problem. I'm keeping dependencies as:

 ```
   "@babel/core": "7.0.0-beta.47",
    "@babel/plugin-proposal-optional-chaining": "7.0.0-beta.47",
    "@babel/preset-env": "7.0.0-beta.47",
    "@babel/preset-stage-2": "7.0.0-beta.47",
```

before babel 7 is stable for react native

**Comment by existentialism** [comment]
@mehulmpt just to clarify: “babel 7 is stable for react native”... it is, it’s just that all your @babel/* dependencies need to fixed to matching versions, and since RN is currently using beta47, you must too.

**Comment by mehulmpt** [comment]
Yeah, I meant to say "before we can fallback to semver for babel 7 packages for react native"

**Comment by schumannd** [comment]
@somonek s answer solved it for me:

before:

```
{
  "presets": ["react-native"],
  "plugins": [
    "transform-decorators-legacy"
  ]
}

```

after:

```
{
  "presets": ["react-native"],
  "plugins": [
    ["@babel/plugin-proposal-decorators", {
      "legacy": true
    }],
  ]
}

```

**Comment by hvaoc** [comment]
@rborn - Can't let folks suffer... 

This is what I am having in my project to make Mobx work with React Native 0.56, hope this helps you all. 

**.babelrc**
```json
{
  "presets": [
    "react-native"
  ],
  "plugins": [
    [
      "@babel/plugin-proposal-decorators",
      {
        "legacy": true
      }
    ]
  ]
}
```

**package.json**

```json
{
  "name": "MyProject",
  "version": "0.0.1",
  "dependencies": {
    "@babel/plugin-proposal-decorators": "7.0.0-beta.47",
    "@babel/runtime": "7.0.0-beta.47",
    "add": "^2.0.6",
    "axios": "^0.18.0",
    "deep-object-diff": "^1.1.0",
    "fast-json-patch": "^2.0.7",
    "flat": "^4.1.0",
    "lodash": "^4.17.10",
    "mobx": "^5.0.3",
    "mobx-react": "^5.2.3",
    "react": "16.4.1",
    "react-dom": "^16.4.1",
    "react-native": "^0.56.0",
    "react-native-code-push": "^5.4.1",
    "react-native-country-picker-modal": "^0.6.2",
    "react-native-firebase": "^4.3.7",
    "react-native-form": "^2.1.2",
    "react-native-gifted-chat": "^0.4.3",
    "react-native-lightbox": "git://github.com/oblador/react-native-lightbox.git#0a37652",
    "react-native-linear-gradient": "^2.4.0",
    "react-native-loading-spinner-overlay": "^0.5.2",
    "react-native-segmented-control-tab": "^3.2.2",
    "react-native-text-input-enhance": "^1.0.11",
    "react-native-typography": "^1.3.0",
    "react-native-vector-icons": "^4.6.0",
    "react-navigation": "^2.6.2",
    "react-navigation-header-buttons": "^2.0.1",
    "styled-components": "^3.3.3",
    "uuid": "^3.3.2",
    "yarn": "^1.9.4"
  },
  "devDependencies": {
    "babel-jest": "23.4.0",
    "babel-preset-react-native": "^5",
    "flow-bin": "^0.76.0",
    "jest": "^23.4.1",
    "jsc-android": "^224109.0.0",
    "react-native-cli": "^2.0.1",
    "react-native-rename": "^2.2.2",
    "react-test-renderer": "16.4.1"
  },
  "jest": {
    "preset": "react-native"
  }
}

```

**Comment by rborn** [comment]
@hvaoc I fixed like this: 
you need to have `.babelrc` split in dev/production like here https://github.com/oblador/react-native-vector-icons/issues/801#issuecomment-409268915

Alsop keep `@babel/*` vesions to `7.0.0-beta.47`

Thanks a lot for your help 🤗

**Comment by xtuc** [comment]
Since no information were provided in the issue, I'm going to close this. Please feel free to re-open a new issue.
